Sources/Networking/Operations/Handling/CustomerInfoResponseHandler.swift

```diff
@@ -32,7 +68,38 @@ class CustomerInfoResponseHandler {
             }
             .mapError(BackendError.networkError)
 
-        completion(result)
+        self.handle(result: result, completion: completion)
+    }
+
+    private func handle(
+        result: Result<CustomerInfo, BackendError>,
+        completion: @escaping CustomerAPI.CustomerInfoResponseHandler
+    ) {
+        guard self.offlineEntitlementsEnabled,
+              result.error?.isServerDown == true,
+              #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *) else {
+            completion(result)
+            return
+        }
+
+        _ = Task<Void, Never> {
+            do {
+                completion(.success(try await self.computeOfflineCustomerInfo()))
```

Comment:
This is something that was only recently added to android but we also need to clear the cached customer info upon returning the offline entitlements customer info: https://github.com/RevenueCat/purchases-android/blob/main/common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/OfflineEntitlementsManager.kt#L74. I believe we are not doing that here?

This is to avoid some possible edge cases where if a user enters offline entitlements mode, and restarts the app, we don't want them to use the cached customer info which might not contain the latest purchase information. Instead, we prefer to wait to try to fetch the customer info from the backend (and return an offline computed customer info from that if backend still down)

Reply:
This is already done! I missed that https://github.com/RevenueCat/purchases-ios/blob/3edef65e8557e6ea648f0a8917f3f15f44054b03/Sources/Identity/CustomerInfoManager.swift#L197